### PR TITLE
feat: statistics API

### PR DESCRIPTION
**Pull request by shadinaif**
feat: statistics API

### CODE REVIEW

File: futurex_openedx_extensions/dashboard/details/learners.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        )
+    ).annotate(
+        has_site_login=Exists(
+            UserSignupSource.objects.filter(
+                user_id=OuterRef('id'),
+                site__in=tenant_sites
+            )
+        )
+    ).filter(
+        Q(courses_count__gt=0) | Q(has_site_login=True)
```
Comment (OmarIthawi) [inline futurex_openedx_extensions/dashboard/details/learners.py:78]: why this is the case?

File: futurex_openedx_extensions/dashboard/details/learners.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+from common.djangoapps.student.models import CourseAccessRole, UserSignupSource
+from django.contrib.auth import get_user_model
+from django.db.models import Count, Exists, OuterRef, Q, Subquery
+from django.db.models.query import QuerySet
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+
+from futurex_openedx_extensions.helpers.tenants import get_course_org_filter_list, get_tenant_site
+
+
+def get_learners_queryset(tenant_ids: List, search_text: str = None) -> QuerySet:
```
Comment (OmarIthawi) [inline futurex_openedx_extensions/dashboard/details/learners.py:15]: This query is a bit slow on production. Any idea how can we make it faster?

**Review by OmarIthawi — APPROVED** [review]
@shadinaif please go ahead and merge this pull request. I wasn't aware it was blocked by review.

File: futurex_openedx_extensions/dashboard/details/learners.py (hunk 2 above)
Comment (shadinaif) [reply futurex_openedx_extensions/dashboard/details/learners.py:15]: I believe the issue is not this particular query. It's the other helper queries, you'll find a few methods with `TODO` to add cache. These will increase the performance significantly as far as I can see

File: futurex_openedx_extensions/dashboard/details/learners.py (hunk 1 above)
Comment (shadinaif) [reply futurex_openedx_extensions/dashboard/details/learners.py:78]: We need the list of users related to the tenant. So, the user must be enrolled in one of the tenant's courses, or have a site login in the tenant's site

File: futurex_openedx_extensions/dashboard/details/learners.py (hunk 2 above)
Comment (shadinaif) [reply futurex_openedx_extensions/dashboard/details/learners.py:15]: if caching didn't fix it, we might need to do something with `UserSignupSource`. We need it to record logins, not just signups. So we can use it to directly filter the user by site.

Currently, there are too many records for users with enrollments in a site, but without a record in `UserSignupSource`

**Pull request merged**